### DISCUSSION TITLE
Plot without ax manipulation for plot composition

### DISCUSSION THREAD

**Discussion by Hendrik1987** [DHI/mikeio#556] [Ideas]
# Background
Most plot functionalities in `mikeio` make default settings for figsize, axis limits and axis ratio. This is super handy for most use cases in order to quickly get the most relevant plot appearance. 

However, sometimes if find that I want to compile multiple mikeio plots in the same `ax`. I can pass the `ax` argument, but still, the latest call of a plot function will define the axis ratio etc..

# Example

I want to plot the outline of an inner domain into the parent domain. The directly available solution does not give the desired result as it that will zoom into the inner domain. Also the title will be governed by the `outline` plot.

```
ax = plt.subplot()
dsReg0.geometry.plot(ax=ax)
dsLoc0.geometry.plot.outline(ax=ax)
```

# My workaround
Currently I achieve what I want like this:

```
def outline_only(self, ax=None):
    linwid = 1.2
    out_col = "0.4"
    for exterior in self.g.boundary_polylines.exteriors:
        ax.plot(*exterior.xy.T, color=out_col, linewidth=linwid)
    for interior in self.g.boundary_polylines.interiors:
        ax.plot(*interior.xy.T, color=out_col, linewidth=linwid)
mikeio.spatial.FM_geometry._GeometryFMPlotter.outline_only = outline_only

ax = plt.subplot()
dsReg0.geometry.plot(ax=ax)
dsLoc0.geometry.plot.outline_only(ax=ax)
```

Monkey patching to a private class is of course not very tidy...

# Desired functionality / idea
I think it could be nice to have the option to only add objects to an axis without manipulating the axis.
Assuming that when `ax` is passed, it is not manipulated might be too simple, so maybe just a `plot_only` argument would do?

Looking forward to other thoughts and ideas :-)

**Comment by ecomodeller** [comment]
Just to be clear on what you want to achieve in this specific case.

You only care about the extent of the model domains?

**Comment by ecomodeller** [comment]
I would do it like this:

```python
import mikeio
import matplotlib.pyplot as plt

msh = mikeio.Mesh("../tests/testdata/odense_rough.mesh")

exterior = msh.geometry.to_shapely().buffer(0) # this is the model domain
subdomain = exterior.centroid.buffer(1000) # this is just to illustrate a possible subdomain

plt.plot(*exterior.exterior.xy, label="Model domain")
plt.plot(*subdomain.exterior.xy, color="red", label="Subdomain")
plt.legend();
```

![image](https://user-images.githubusercontent.com/614215/225005992-6c2f59b6-d5c3-4bb7-97f1-fc304ff60f48.png)

**Reply by Hendrik1987** [reply]
Thanks. A good solution.

It was probably not clear above, but I actually have two `.dfsu` files / `geometries` as input for this. For the outer model domain I would like to show the mesh, which can be achieved with your solution, too.

I just thought that some shorthand similar to this might be a nice feature?
```
ax = plt.subplot()
ds_domain.geometry.plot(ax=ax)
ds_subdomain.geometry.plot.outline(ax=ax, plot_only=True)
```

IMHO it could be nice to easily use the core plot functionalities.

xarray for example alows to plot a [map](https://docs.xarray.dev/en/stable/user-guide/plotting.html#maps) using cartopy. (Although this is more about passing `kwargs` than manipulating the axis).

**Reply by ecomodeller** [reply]
Agree

**Reply by Hendrik1987** [reply]
I could have a go at a PR. Just wanted to discuss ideas here first as to what viable concept could be. As it would probably be nice to role it out accross all plot functions :-)
